server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataTests.java

```diff
@@ -794,4 +794,12 @@ public static void assertMultiField(Map<String, Object> properties, String field
             "    }\n" +
             "  }\n" +
             "}";
+
+    public void testTransientSettingsOverridePersistentSettings() {
+        MetaData metaData = MetaData.builder()
+            .persistentSettings(Settings.builder().put("key", "persistent-value").build())
+            .transientSettings(Settings.builder().put("key", "transient-value").build()).build();
+
+        assertThat(metaData.settings().get("key"), equalTo("transient-value"));
```

Comment:
I did not check but I think it should be possible to test this with a setting object instead doing a raw get on the settings instance; that is `Settings#get(String)` versus `Setting#get(settings)`. Would you please write the test that way? It is more realistic.

Reply:
Yes, this was possible. I pushed 1ec65d0.